### ISSUE TITLE
Split of the Unseen entities

### ISSUE THREAD

**Issue by lbq8942** [JinheonBaek/GEN#6]
Thank you for your excellent work!

Unseen entities are split into (meta-)train/val/test. In your code, i found that the triplets in `self.meta_train_task_entity_to_triplets`, `self.meta_val_task_entity_to_triplets` and  `self.meta_test_task_entity_to_triplets` are disjoint respectively. Is this just a coincidence or a must? Because i have a question, if entity **s** is divided into train and entity **o**  into test, the triplet (s,r,o) belongs to which? In this case, the triplets in train and test will not be disjoint.

**Comment by JinheonBaek** [comment]
Thanks for your interest in our work!

Yes, the task entities in training, validation, and test sets must be disjoint. 
For your next question, we also make sure that a certain subject entity in the training set should not appear as the object entity in the test set.
